packages/bp/src/core/dialog/dialog-engine.ts

```diff
@@ -304,6 +305,10 @@ export class DialogEngine {
 
     this.fillContextForTransition(event, { currentFlow, currentNode, nextFlow: timeoutFlow, nextNode: timeoutNode })
 
+    if (event.hasFlag(WellKnownFlags.FORCE_PERSIST_STATE)) {
+      await this.stateManager.persist(event, false)
```

Comment:
I don't think the dialog engine should be aware of the state manager. Also, I don't see what that flag's purpose is, how is it related to the timeout ?

Reply:
To be honest, I'm not really sure, it's implemented both behind the flag and without it during event processing at the processEvent function in the DecisionEngine. Looking at the code, it looks like that it is supposed to be used when you want to skip the dialog engine or the decision engine chose to elect a QnA but you still want to save the state change, is my assumption correct? My guess was that maybe changing the default behavior could break something somewhere, that is why I choose to keep this behind a flag since, for my use case, there were no side effects.

Reply:
@allardy I'm going to move this code logic to the janitor, as suggested

Reply:
@allardy I'm going to move this code logic to the janitor, as suggested